pkg/controller/userdatasecret/userdatasecret_controller.go

```diff
@@ -0,0 +1,217 @@
+		// ignore create event for userData secret creation
+		CreateFunc: func(e event.CreateEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			secretName := e.MetaNew.GetName()
+			if secretName == userDataSecret {
+				return true
+			}
+			return false
+		},
+		// ignore delete event, as we will get an update event while deleting
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			secretName := e.Meta.GetName()
+			if secretName == userDataSecret {
+				return true
+			}
+			return false
+		},
+	}
+
+	// Watch for changes to primary resource UserDataSecret
+	err = c.Watch(&source.Kind{Type: &corev1.Secret{ObjectMeta: meta.ObjectMeta{Namespace: "openshift-machine-api"}}}, &handler.EnqueueRequestForObject{}, predicateFilter)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// blank assignment to verify that ReconcileUserDataSecret implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileUserDataSecret{}
+
+// ReconcileUserDataSecret reconciles a UserDataSecret object
+type ReconcileUserDataSecret struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+	// signer is a signer created from the user's private key
+	signer ssh.Signer
+}
+
+// Reconcile reads that state of the cluster for a UserDataSecret object and makes changes based on the state read
+// and what is in the UserDataSecret.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileUserDataSecret) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling UserDataSecret")
+
+	userData, err := r.createUserDataSecret()
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "error creating user data secret")
+
+	}
+	// Fetch UserData instance
+	err = r.client.Get(context.TODO(), kubeTypes.NamespacedName{Name: userData.Name, Namespace: userData.Namespace}, &corev1.Secret{})
+	if err != nil && k8sapierrors.IsNotFound(err) {
+		// Secret is deleted
+		reqLogger.Info("Creating a new Secret", "Secret.Namespace", userData.Namespace, "Secret.Name", userData.Name)
+		err = r.client.Create(context.TODO(), userData)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		// Secret created successfully - don't requeue
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		return reconcile.Result{}, err
+	} else {
+		// secret is updated
+		reqLogger.Info("Updating the secret", "Secret.Namespace", userData.Namespace, "Secret.Name", userData.Name)
```

Comment:
Updating the secret -> Updating userdata secret

Reply:
`reqLogger` should anynow show the name and namespace. It needed be repeated here. Also `Restoring...` would be more informative.